src/decision/dMCDA.jl

```diff
@@ -45,6 +49,53 @@ function mcda_methods()
     ]
 end
 
+"""
+    mcda_method_names()::Vector{String}
+
+List of MCDA method names.
+"""
+function mcda_method_names()::Vector{String}
+    return [
+        "Cocoso",
+        "Mairca",
+        "Moora",
+        "Piv",
+        "Vikor"
+    ]
+end
+
+"""
+    mcda_method_encoding(mcda_name::String)::Union{Int64, Nothing}
+
+Find the scenario encoding of the given mcda method. Throws `ArgumentError` if method not 
+found.
+"""
+function mcda_method_encoding(mcda_name::String)::Int64
+    method_idx = findfirst(mcda_method_names() .== mcda_name)
+    if isnothing(method_idx)
+        msg = "Given MCDA method $(mcda_name) is not in list of mcda methods. "
+        msg *= "Possible MCDA methods are $(mcda_method_names())"
+        throw(ArgumentError(msg))
+    end
+    return method_idx
+end
+
+"""
+    decision_method_encoding(method_name::String)::Int64
+
+Get the integer encoding for the type of intervention to be used in the scenario dataframe.
+"""
+function decision_method_encoding(method_name::String)::Int64
+    if method_name == "counterfactual"
+        return COUNTERFACTUAL_SCEN_ENCODING
+    elseif method_name == "unguided"
+        return UNGUIDED_SCEN_ENCODING
+    else
+        return mcda_method_encoding(method_name)
+    end
+    return nothing
```

Comment:
```suggestion
    if method_name == "counterfactual"
        return COUNTERFACTUAL_SCEN_ENCODING
    elseif method_name == "unguided"
        return UNGUIDED_SCEN_ENCODING
    end
    
    return mcda_method_encoding(method_name)
```

This method only works as currently written as you always return an Int64 (it never hits the `return nothing`).
Otherwise, it would raise an error as `nothing` can't be casted to `Int64`.